### PR TITLE
[msal-browser] Add option to use Secure Cookies to cache configuration

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:

- Adds the `secureCookies` flag to `CacheOptions` type
- Adds logic in `setItemCookie` to check said flag and add the "Secure" flag to the cookie string
- Adds unit tests for the `secureCookies` configuration option
- Adds documentation on the `secureCookies` configuration option

- Addresses #2854

### CODE REVIEW

File: lib/msal-browser/test/config/Configuration.spec.ts [hunk 1]
```diff
@@ -171,6 +174,8 @@ describe("Configuration.ts Class Unit Tests", () => {
         expect(newConfig.cache.cacheLocation).to.be.eq("localStorage");
         expect(newConfig.cache.storeAuthStateInCookie).to.be.not.null;
         expect(newConfig.cache.storeAuthStateInCookie).to.be.true;
+        expect(newConfig.cache.secureCookies).to.be.not.null;
+        expect(newConfig.cache.secureCookies).to.be.true;
```
Comment (jasonnutter) [inline lib/msal-browser/test/config/Configuration.spec.ts:178]: Just need the second one, first one is not necessary.

Comment (jasonnutter) [reply lib/msal-browser/test/config/Configuration.spec.ts:178]: Although I see all the other assertions make those same two checks. Not a big deal, ultimately.

File: lib/msal-browser/src/cache/BrowserCacheManager.ts [hunk 2]
```diff
@@ -506,6 +506,10 @@ export class BrowserCacheManager extends CacheManager {
             cookieStr += `expires=${expireTime};`;
         }
 
+        if(this.cacheConfig.secureCookies) {
```
Comment (jasonnutter) [inline lib/msal-browser/src/cache/BrowserCacheManager.ts:509]:
```suggestion
        if (this.cacheConfig.secureCookies) {
```

Comment (jasonnutter) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts:509]: Suprised linting didnt catch that.

**Review by jasonnutter — APPROVED** [review]

File: lib/msal-browser/test/config/Configuration.spec.ts [hunk 3]
```diff
@@ -30,6 +30,8 @@ describe("Configuration.ts Class Unit Tests", () => {
         expect(emptyConfig.cache.cacheLocation).to.be.eq("sessionStorage");
         expect(emptyConfig.cache.storeAuthStateInCookie).to.be.not.null.and.not.undefined;
         expect(emptyConfig.cache.storeAuthStateInCookie).to.be.false;
+        expect(emptyConfig.cache.secureCookies).to.be.not.null.and.not.undefined;
```
Comment (hectormmg) [inline lib/msal-browser/test/config/Configuration.spec.ts:33]:
```suggestion
```

File: lib/msal-browser/test/config/Configuration.spec.ts [hunk 4]
```diff
@@ -171,6 +174,8 @@ describe("Configuration.ts Class Unit Tests", () => {
         expect(newConfig.cache.cacheLocation).to.be.eq("localStorage");
         expect(newConfig.cache.storeAuthStateInCookie).to.be.not.null;
         expect(newConfig.cache.storeAuthStateInCookie).to.be.true;
+        expect(newConfig.cache.secureCookies).to.be.not.null;
```
Comment (hectormmg) [inline lib/msal-browser/test/config/Configuration.spec.ts:177]:
```suggestion
```

File: lib/msal-browser/test/config/Configuration.spec.ts (hunk 1 above)
Comment (hectormmg) [reply lib/msal-browser/test/config/Configuration.spec.ts:178]: Removing this assertion and the equivalent in the previous test. I agree they are redundant.

File: lib/msal-browser/src/cache/BrowserCacheManager.ts [hunk 5]
```diff
@@ -752,7 +756,8 @@ export class BrowserCacheManager extends CacheManager {
 export const DEFAULT_BROWSER_CACHE_MANAGER = (clientId: string, logger: Logger) => {
     const cacheOptions = {
         cacheLocation: BrowserCacheLocation.MemoryStorage,
-        storeAuthStateInCookie: false
+        storeAuthStateInCookie: false,
+        secureCookies: false
```
Comment (pkanher617) [inline lib/msal-browser/src/cache/BrowserCacheManager.ts:760]: Should we separate cookie config into its own object?

Comment (hectormmg) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts:760]: I'd be interested in hearing others' opinions. If we foresee adding more cookie configuration in the future, then I'd definitely vote we make a dedicated object under CacheConfig.

Comment (jasonnutter) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts:760]: I'd say in v3 lets do that, will be a bit weird given we wont be able to put storeAuthStateInCookie in it now.

Comment (pkanher617) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts:760]: sounds good.

**Review by pkanher617 — APPROVED** [review]

File: lib/msal-browser/src/cache/BrowserCacheManager.ts (hunk 5 above)
Comment (sameerag) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts:760]: Agree to make this change this in a major version.

**Review by sameerag — APPROVED** [review]

**Comment by jasonnutter** [comment]
@hectormmg Good to merge?

**Comment by AbhaysinghBhosale** [comment]
Migrating from msal 1.3.2  to msal-browser impact on code and need to move to auth code flow. Cant we provide this option on msal itself for those who are using msal with explicit flow?

**Pull request merged**